network/secondary_endpoint_client_linux.go

```diff
@@ -0,0 +1,150 @@
+	// ip route del <multitenant ip> dev <ethX>
+	// Deleting the route set up for routing the incoming packets to pod
+	for ifName, ifInfo := range ep.SecondaryInterfaces {
+		log.Printf("[net] deleting routes on %s: %+v", ifName, ifInfo.Routes)
+		if err := deleteRoutes(client.netlink, client.netioshim, ifName, ifInfo.Routes); err != nil {
+			log.Printf("[net] Failed to delete routes %+v on %s: %v", ifInfo.Routes, ifName, err)
+		}
+	}
+}
+
+func (client *SecondaryEndpointClient) MoveEndpointsToContainerNS(epInfo *EndpointInfo, nsID uintptr) error {
+	// Move the container interface to container's network namespace.
+	log.Printf("[net] Setting link %v netns %v.", epInfo.IfName, epInfo.NetNsPath)
+	if err := client.netlink.SetLinkNetNs(epInfo.IfName, nsID); err != nil {
+		return newErrorSecondaryEndpointClient(err)
+	}
+
+	return nil
+}
+
+func (client *SecondaryEndpointClient) SetupContainerInterfaces(epInfo *EndpointInfo) error {
+	log.Printf("[net] Setting link %v state up.", epInfo.IfName)
+
+	return client.netlink.SetLinkState(epInfo.IfName, true)
+}
+
+func (client *SecondaryEndpointClient) ConfigureContainerInterfacesAndRoutes(epInfo *EndpointInfo) error {
+	if err := client.netUtilsClient.AssignIPToInterface(epInfo.IfName, epInfo.IPAddresses); err != nil {
+		return newErrorSecondaryEndpointClient(err)
+	}
+
+	ifInfo, exists := client.ep.SecondaryInterfaces[epInfo.IfName]
+	if !exists {
+		return newErrorSecondaryEndpointClient(errors.New(epInfo.IfName + " does not exist"))
+	}
+
+	if epInfo.IsDefaultInterface {
+		// add route for virtualgwip
+		// ip route add 169.254.1.1/32 dev ethX
+		virtualGwIP, virtualGwNet, _ := net.ParseCIDR(virtualGwIPString)
+		routeInfo := RouteInfo{
+			Dst:   *virtualGwNet,
+			Scope: netlink.RT_SCOPE_LINK,
+		}
+		if err := addRoutes(client.netlink, client.netioshim, epInfo.IfName, []RouteInfo{routeInfo}); err != nil {
+			return newErrorSecondaryEndpointClient(err)
+		}
+
+		// TO-DO: will default routes need to be cleaned up even though they're added in pod netns?
+		// looks like interface goes back to default state (down without routes) after deleting pod
+		ifInfo.Routes = append(ifInfo.Routes, routeInfo)
+
+		// ip route add default via 169.254.1.1 dev ethX
+		_, defaultIPNet, _ := net.ParseCIDR(defaultGwCidr)
+		dstIP := net.IPNet{IP: net.ParseIP(defaultGw), Mask: defaultIPNet.Mask}
+		routeInfo = RouteInfo{
+			Dst: dstIP,
+			Gw:  virtualGwIP,
+		}
+		if err := addRoutes(client.netlink, client.netioshim, epInfo.IfName, []RouteInfo{routeInfo}); err != nil {
+			return newErrorSecondaryEndpointClient(err)
+		}
+
+		// TO-DO: will default routes need to be cleaned up even though they're added in pod netns?
+		// looks like interface goes back to default state (down without routes) after deleting pod
+		ifInfo.Routes = append(ifInfo.Routes, routeInfo)
+	} else if err := addRoutes(client.netlink, client.netioshim, epInfo.IfName, epInfo.Routes); err != nil {
```

Comment:
can skip `else if` as like above comment

Reply:
we can, but at this moment we only expect the non default interface to have routes and this makes it clear

Reply:
updated